src/backend/expungeservice/models/charge_types/duii.py

```diff
@@ -9,17 +10,18 @@ def _default_type_eligibility(self):
         if not self.disposition:
             return TypeEligibility(EligibilityStatus.NEEDS_MORE_ANALYSIS, reason='Further Analysis Needed - No disposition')
 
-        elif 'diverted' in self.disposition.ruling.lower():
-            return TypeEligibility(EligibilityStatus.INELIGIBLE, reason='Ineligible under 137.225(8)(b)')
+        elif self.disposition.status == DispositionStatus.DIVERTED:
+            return TypeEligibility(EligibilityStatus.INELIGIBLE, reason='137.225(8)(b) - Diverted DUII charges are ineligible')
 
-        elif self.acquitted():
-            return TypeEligibility(EligibilityStatus.NEEDS_MORE_ANALYSIS, reason='Further Analysis Needed')
+        elif self.convicted():
 
-        else:
             """
-            TODO: other dispositions than "convicted" are possible and are not handled here.
-            See issue: https://github.com/codeforpdx/recordexpungPDX/issues/601
-
             TODO: reason can be expanded to "137.225(7)(a) - Traffic offenses are ineligible"
             """
-            return TypeEligibility(EligibilityStatus.INELIGIBLE, reason='Ineligible under 137.225(7)(a)')
+            return TypeEligibility(EligibilityStatus.INELIGIBLE, reason='137.225(7)(a) - Traffic offenses are ineligible')
+
+        elif self.acquitted():
+            return TypeEligibility(EligibilityStatus.NEEDS_MORE_ANALYSIS, reason='Further Analysis Needed - Dismissed charge may have been Diverted')
+
+        else:
+            return TypeEligibility(EligibilityStatus.NEEDS_MORE_ANALYSIS, reason='Further Analysis Needed - Unrecognized ruling')
```

Comment:
For now, it looks like the only reason we are `__convicted`/`acquitted` in `charge.py` is because of changes in these files. This may have side-effects on the relationship between other charge subclasses and the expunger (and in fact most likely they are "good"/bug-fixing side-effects). However, I think we need to be more explicit about those side-effects as any future reader of this PR may think that `__convicted`/`acquitted` changes are solely for this `duii.py` file.

Since we have "full coverage" of the DispositionStatus Enum here, we should just do a "switch/case" imitation like the following:

```python
if self.disposition:
    dismissed_type_eligibility = TypeEligibility(EligibilityStatus.NEEDS_MORE_ANALYSIS, reason='Further Analysis Needed - Dismissed charge may have been Diverted')
    unknown_type_eligibility = ...
    cases = {
        DispositionStatus.CONVICTED: TypeEligibility(EligibilityStatus.INELIGIBLE, reason='137.225(7)(a) - Traffic offenses are ineligible')
        DispositionStatus.DISMISSED: dismissed_type_eligibility
        DispositionStatus.NO_COMPLAINT: dismissed_type_eligibility
        DispositionStatus.DIVERTED:  ...
        DispositionStatus.UNKNOWN: unknown_type_eligibility
    }
    assert len(cases) == len(DispositionStatus) # So that if someone adds something to DispositionStatus, it won't automatically go to the Unknown case.
    return cases.get(self.disposition.status, unknown_type_eligibility) 
else:
     ...
```

And then I suggest a separate PR for the `__convicted`/`acquitted` changes that adds comments how it affects the expunger (or how it doesn't).